src/org/labkey/remoteapi/plate/PlateSetParams.java

```diff
@@ -0,0 +1,92 @@
+package org.labkey.remoteapi.plate;
+
+
+import org.json.JSONObject;
+
+public class PlateSetParams
+{
+    private String _name = null;
+    private String _description = null;
+    private String _type = null;
+    private Integer _id = null;
+    private Integer _parentPlateSetId = null;
+
+    public PlateSetParams()
+    {
+    }
+
+    public PlateSetParams(JSONObject json)
+    {
+        _name = json.getString("name");
+        _description = json.optString("description", null);
+        _type = json.optString("type", null);
+        _id = json.getInt("rowId");
+        _parentPlateSetId = json.optIntegerObject("parentPlateSetId", null);
+    }
+
+    public JSONObject toJSON()
+    {
+        JSONObject json = new JSONObject();
+        if (_name != null)
+            json.put("name", _name);
+        if (_description != null)
+            json.put("description", _description);
+        if (_type != null)
+            json.put("type", _type);
+        if (_id != null)
+            json.put("rowId", _id);
+        if (_parentPlateSetId != null)
+            json.put("parentPlateSetId", _parentPlateSetId);
+        return json;
```

Comment:
These `null` checks actually aren't necessary. Much to our chagrin, `JSONObject` doesn't allow `null` values. A `put` with a `null` value actually removes any existing entry and does nothing if there is no existing entry.
```suggestion
        JSONObject json = new JSONObject();
        json.put("name", _name);
        json.put("description", _description);
        json.put("type", _type);
        json.put("rowId", _id);
        json.put("parentPlateSetId", _parentPlateSetId);
        return json;
```

Reply:
Thank you! (I didn't know that, it feels good to learn something new)